NightscoutServiceKit/Extensions/DeviceStatus.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  DeviceStatus.swift
+//  NightscoutServiceKit
+//
+//  Created by Darin Krauss on 10/17/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import HealthKit
+import LoopKit
+import NightscoutUploadKit
+
+extension DeviceStatus {
+
+    init(storedDosingDecision: StoredDosingDecision) {
+        var iob: IOBStatus?
+        if let insulinOnBoard = storedDosingDecision.insulinOnBoard {
+            iob = IOBStatus(timestamp: insulinOnBoard.startDate, iob: insulinOnBoard.value)
```

Comment:
[nit] would it be useful to map `InsulinValue` to `IOBStatus` using an extension (e.g., `storedDosingDecision.insulinOnBoard.iobStatus: IOBStatus?`). If yes, consider this for the other `storedDosingDecision` properties below as well.

Reply:
Yes. Looking at this again (and the other convenience initializers) I think I'm going to reverse the logic. Rather than having a convenience initializer for the new type I'm going to add an extension on the old type with a property to generate the new type (and each part of the new type as you describe).